scripts/launch-platform.sh

```diff
@@ -0,0 +1,267 @@
+    multipass exec "${VMNAME}" -- /bin/bash /home/ubuntu/setup-multipass.sh --mode 'local' --name "${VMNAME}" --branch "${BRANCH}"
+}
+
+
+function setup_cache {
+    sudo mkdir -p /cache/downloads /cache/yarn /cache/apt-archives
+    sudo chown ${UID}:0 /cache/{downloads,yarn}
+    chmod 770  /cache/{downloads,yarn}
+
+    sudo chown root:root /cache/apt-archives
+    sudo chmod 755 /cache/apt-archives
+    sudo rm -r /var/cache/apt/archives
+    sudo ln -s  /cache/apt-archives /var/cache/apt/archives || true
+}
+
+
+function cleanup_multipass {
+    if ! which multipass >/dev/null; then
+        echo 'Please install multipass; see https://multipass.run/ for more details.'
+        exit 1
+    fi
+
+    multipass delete "${VMNAME}"
+    multipass purge
+}
+
+function parse_args {
+    while [ "$#" -ge 1 ]; do
+        arg="$1"
+        case $arg in
+        -b | --branch)
+            shift
+            BRANCH="$1"
+            shift
+            ;;
+        --cache)
+            shift
+            CACHE="$(realpath "$1")"
+            shift
+            ;;
+        --cpus)
+            shift
+            CPUS="$1"
+            shift
+            ;;
+        -d | --debug)
+            DEBUG=1
+            shift
+            ;;
+        -h | --help)
+            MODE=help
+            shift
+            ;;
+        --mode)
+            shift
+            MODE="$1"
+            shift
+            ;;
+        -n | --name)
+            shift
+            VMNAME="$1"
+            shift
+            ;;
+        --repo)
+            shift
+            REPO="$(realpath "$1")"
+            shift
+            ;;
+
+        *)
+            echo "Unrecoginzed command line argument '$arg';  try '$0 --help'"
+            exit 1
+            ;;
+        esac
+    done
+
+    if [[ -n "$DEBUG" ]]; then
+        set -x
+    fi
+
+    VMNAME="${VMNAME:-dash}"
+    if [[ -n "${REPO}" ]] && [[ -n "${BRANCH}" ]] ; then
+      echo 'ERROR: --repo and --branch are mutually exclusive, please use only one of them.'
+      exit 1
+    fi
+
+    BRANCH="${BRANCH:-master}"
+    MODE="${MODE:-multipass}"
+
+}
+
+# Download with caching
+# Usage: download SRC_URL FILENAME
+function download {
+    FILENAME="$2"
+    if [[ -z "${FILENAME}" ]] ; then
+        FILENAME="$(basename "$1")"
+    fi
+    if [[ ! -f "${DOWNLOADS_CACHE}/${FILENAME}" ]] ; then
+        wget --continue -O "${DOWNLOADS_CACHE}/${FILENAME}" "$1"
+    fi
+}
+
+
+
+function install_deps {
+    export DEBIAN_FRONTEND=noninteractive
+    export TZ=UTC
+    sudo apt-get update
+    sudo apt-get install -y tzdata
+    sudo ln -fs /usr/share/zoneinfo/UTC /etc/localtime
+    sudo dpkg-reconfigure --frontend noninteractive tzdata
+    sudo apt-get --yes install build-essential \
+        pkg-config \
+        wget \
+        cmake \
+        git \
+        python3 \
+        ca-certificates \
+        curl \
+        gnupg \
+        lsb-release \
+        libssl-dev \
+        clang
```

Comment:
I think libzmq3-dev would be also good to have in the system. There is a floating issue when you try to pull yarn dependencies, it happens from time to time.

`➤ YN0009: │ zeromq@npm:5.2.8 couldn't be built successfully (exit code 1, logs can be found here: /tmp/xfs-681aae58/build.log)
/usr/bin/ld: cannot find ./Release/../../zmq/lib/libzmq.a: No such file or directory
collect2: error: ld returned 1 exit status
make: *** [zmq.target.mk:141: Release/obj.target/zmq.node] Error 1
make: Leaving directory '/home/dashmate/platform/.yarn/unplugged/zeromq-npm-5.2.8-213a0f74bc/node_modules/zeromq/build'
`